### PR TITLE
Recursively look for oc.json, starting from componentsDir 

### LINKED ISSUE #219: oc dev looks for local oc.json but could do better than that
Basically,  when 'oc-components' is a folder of components, 
* `cd oc-components && oc dev . 3030` works (if oc.json is there)
* just `oc dev oc-component 3030` doesn't (because it looks for oc.json in that folder).

The cli should try to look for oc.json inside the folder and then in the parent if not found.

### PR DESCRIPTION
**Pull request by matteofigus**
Fixes #219

### CODE REVIEW

File: src/cli/domain/get-mocked-plugins.js [hunk 1]
```diff
@@ -25,11 +25,12 @@ var registerStaticMocks = function(mocks, logger){
   });
 };
 
-var registerDynamicMocks = function(mocks, logger){
+var registerDynamicMocks = function(mocks, ocJsonPath, logger){
```
Comment (mattiaerre) [inline src/cli/domain/get-mocked-plugins.js]: I think you should rename this parameter into `ocJsonLocation` as per the way you invoke the method below in line 97

File: src/cli/domain/get-mocked-plugins.js [hunk 2]
```diff
@@ -53,23 +54,47 @@ var registerDynamicMocks = function(mocks, logger){
...
+
+      return findPath(parentDir, fileName);
+    }
+  }
+  
+  return fileToResolve;
+};
+
+module.exports = function(logger, componentsDir){
+  componentsDir = path.resolve(componentsDir || '.');
```
Comment (mattiaerre) [inline src/cli/domain/get-mocked-plugins.js]: so if you do not pass the `componentsDir` parameter you set it to '`.` correct?

File: test/unit/cli-domain-get-mocked-plugins.js [hunk 3]
```diff
@@ -2,45 +2,123 @@
...
+        expect(result.length).to.equal(3);
+      });
+    });
+
     describe('when oc.json is missing', function(){
       var result;
       beforeEach(function(){
         initialise({ existsSync: sinon.stub().returns(false) });
-        result = getMockedPlugins(console);
+        result = getMockedPlugins(console, '/root/components/');
```
Comment (mattiaerre) [inline test/unit/cli-domain-get-mocked-plugins.js]: I see that now all the tests are passing the path parameter to the `getMockedPlugins` method; I think we should have tests w/o the second parameter as well what do you think?

File: src/cli/domain/get-mocked-plugins.js [hunk 4]
```diff
@@ -53,23 +54,47 @@ var registerDynamicMocks = function(mocks, logger){
...
+
   if(!content.mocks || !content.mocks.plugins){
     return plugins;
   }
 
   logger.log(colors.yellow(strings.messages.cli.REGISTERING_MOCKED_PLUGINS));
 
   plugins = plugins.concat(registerStaticMocks(content.mocks.plugins.static, logger));
-  plugins = plugins.concat(registerDynamicMocks(content.mocks.plugins.dynamic, logger));
+  plugins = plugins.concat(registerDynamicMocks(content.mocks.plugins.dynamic, ocJsonLocation, logger));
```
Comment (mattiaerre) [inline src/cli/domain/get-mocked-plugins.js]: I am also thinking about preserve the same "pattern" between the methods (registerStaticMocks and registerDynamicMocks) such as adding the extra parameter either at the very beginning or after "logger" but not in between 😅 

`registerDynamicMocks(ocJsonLocation, content.mocks.plugins.dynamic, logger)`

or

`registerDynamicMocks(content.mocks.plugins.dynamic, logger, ocJsonLocation)`

File: src/cli/domain/get-mocked-plugins.js (hunk 2 above)
Comment (matteofigus) [inline src/cli/domain/get-mocked-plugins.js]: correct.

File: test/unit/cli-domain-get-mocked-plugins.js (hunk 3 above)
Comment (matteofigus) [inline test/unit/cli-domain-get-mocked-plugins.js]: I can do it.

File: src/cli/domain/get-mocked-plugins.js (hunk 1 above)
Comment (matteofigus) [inline src/cli/domain/get-mocked-plugins.js]: ok

**Comment by mattiaerre** [comment]
LGTM 👍  (squashmerge)

**Pull request merged**